plugins/sentinelone/komand_sentinelone/tasks/monitor_activities_and_events/task.py

```diff
@@ -0,0 +1,327 @@
+        else:
+            response = self.connection.client.get_threats(query_params, API_VERSION, True, False)
+        try:
+            status_code = response.status_code
+            if response.status_code in [401, 403]:
+                self.logger.info(f"Received status code {status_code} for {log_type} query")
+            response_handler(response, allowed_status_codes=[401, 403])
+        except PluginException as error:
+            raise ApiException(cause=error.cause, assistance=error.assistance, status_code=status_code, data=response)
+        logs, next_page_cursor, total_forbidden_responses = self.extract_query_response(
+            response, total_forbidden_responses
+        )
+        new_logs_last_timestamp = self.get_latest_timestamp(logs, last_log_timestamp)
+        state[timestamp_key] = new_logs_last_timestamp
+        state[page_token_key] = next_page_cursor
+        if next_page_cursor:
+            self.logger.info(f"New next page cursor received for {log_type}: {next_page_cursor}")
+        return logs, total_forbidden_responses
+
+    def get_query_params(self, log_type: str, last_log_timestamp: str, current_run_timestamp: str, cursor: str) -> Dict:
+        """
+        Generate query parameters based on log type and availability of pagination cursor
+        """
+        timestamp_name = "createdAt"
+        if log_type == EVENTS_LOGS:
+            timestamp_name = "eventTime"
+        query_params = {
+            "limit": LOGS_PAGE_LIMIT,
+            f"{timestamp_name}__gt": last_log_timestamp,
+            f"{timestamp_name}__lte": current_run_timestamp,
+            "sortBy": timestamp_name,
+        }
+        if cursor:
+            query_params["cursor"] = cursor
+        return query_params
+
+    def extract_query_response(
+        self, response: requests.Response, total_forbidden_responses: int
+    ) -> Tuple[List[Dict], str]:
+        """
+        Return the log sand the pagination cursor from a query response
+        """
+        if response.status_code in [HTTPStatusCodes.UNAUTHORIZED, HTTPStatusCodes.FORBIDDEN]:
+            total_forbidden_responses += 1
+            return [], None, total_forbidden_responses
+        response_json = extract_json(response)
+        logs = response_json.get("data", [])
+        next_cursor = response_json.get("pagination", {}).get("nextCursor")
+        return logs, next_cursor, total_forbidden_responses
+
+    def get_latest_timestamp(self, logs, latest_timestamp_string) -> Optional[str]:
+        """
+        Return the latest timestamp from a sorted list of logs. If no logs exist, use the last provided timestamp
+        """
+        # If no logs returned, return existing timestamp or None if never previously set
+        if len(logs) == 0:
+            if latest_timestamp_string == STARTING_TIMESTAMP:
+                return None
+            return latest_timestamp_string
+        new_timestamp_string = logs[-1].get("createdAt")
```

Comment:
if we've qeuried for event logs the timestamp name in the query is `eventTime` is it ok to use `createdAt` here?

Reply:
What is current might result in different times. Updated to check if event time exists first.